### DISCUSSION TITLE
Impact of running CLM on FV vs SE grid

### DISCUSSION THREAD

**Discussion by adagj** [NorESMhub/noresm3_dev_simulations#18] [CLM-FATES grid and resolution]
Hi @mvertens and Rosie (why can't I tag Rosie?)
Great if you post some updates about the trigrid versus dual grid simulations here, and maybe indicate where to look at the diagnostics.
Thanks!

**Comment by mvertens** [comment]
@adagj -I'm not sure what should be done for CLM diagnostics. The atm comparison diagnostics for the AMIP configuration are already here https://github.com/NorESMhub/noresm_simulations/issues/13.
But its not clear to me what to do about CLM diagnostics since we can only produce this for the FV output on our platforms.
How do I run the CLM diagnostics on the FV simulation? 
Also - I have just added @rosiealice as a collaborator.

**Comment by adagj** [comment]
The CLM diagnostics on the FV grid can be generated using diag_run I guess. I was more thinking of informing the community about these test simulations and decisions, but #13 is actually doing that. Thanks! 
And thanks for adding Rosie :-)

**Comment by mvertens** [comment]
Here is a brief summary of the issue. 

First a set of definitions:

**maps:**
```
mapo2l - first order conservative mapping of ocn->lnd
mapl2a - first order conservative mapping of lnd->atm
mapo2a - first order conservative mapping of ocn->atm
```

**ocean grid:**
```
ocean mask on ocean grid:
mask_o - the mask on the ocean grid 
(note that this is the same as the ocean fraction on the ocean grid - no partial land)
ifrac_o = ice fraction on ocean grid
ofrac_o = omask_o - ifrac_o
(note that ifrac_o changes dynamically)
```

**land grid**
```
land fraction on land grid:
lfrin_l: land fraction on the land grid computed from mapo2l(1-mask_o)
```

**atm grid:**
```
fractions on atm grid (this is the tricky part)
ifrac_a: ice fraction on the atm grid mapo2a(ifrac_i)  (this changes dynamically)
ofrac_a:  ocean fraction on the atm grid computed from mapo2a(ofrac_o)
lfrac_a: land fraction on the atm grid computed from 1 - ofrac_a - ifrac_a
lfrin_a: land fraction on the atm grid computed from mapl2a(lfrin_l)
NOTE: for a bigrid (atm/lnd on same grid) - lfrin_a and lfrac_a are effectively the same 
for a trigrid they are not
```

**Flux x is generated on ice, ocn and land grids (e.g. sensible heat flux):**
```
x_ll : flux x generated on land grid by land
x_ii: flux x generated on  ice grid by ice
x_oo: flux x generated on ocn grid by ocean

x_la:  mapl2a(x_ll)
x_ia: map_i2a(x_ii)
x_oa: map_o2a(x_oo)
```

**merging to atm for bigrid**
```
X_a = X_l*lfrac_a + X_i*lfrac_i + X_o*lfrac_o
```

**merging to atm for trigrid**
```
1. X_a = X_l*lfrac_a + X_i*lfrac_i + X_o*lfrac_o (what I think)
or 
2. X_a = X_l*lfrin_a + X_i*lfrac_i + X_o*lfrac_o (E3SM)
```

Problem appears in the budget table that is generated by default for coupled runs:
In both cases the compset is
**COMPSET: 1850_CAM%DEV%LT%NORESM%CAMoslo_CLM51%SP_CICE_BLOM%ECO_MOSART_DGLC%NOEVOLVE_SWAV_SESP**
The trigrid grid is
**GRID: a%ne30np4.pg3_l%0.47x0.63_oi%tnx1v4_r%r05_w%null_z%null_g%gris4_m%tnx1v4**
The bigrid grid is 
**GRID: a%ne30np4.pg3_l%ne30npr4.pg3_oi%tnx1v4_r%r05_w%null_z%null_g%gris4_m%tnx1v4**

Below is the annual budget table for the trigrid run:
<img width="1237" alt="Screenshot 2024-08-07 at 2 14 16 PM" src="https://github.com/user-attachments/assets/143c6b0b-d5ee-4b62-8951-c761819cf336">

Below are two annual budget table for the bigrid run:
<img width="1234" alt="Screenshot 2024-08-07 at 11 24 13 AM" src="https://github.com/user-attachments/assets/055458cd-49ad-49f3-b768-37e15f51d25e">

<img width="1229" alt="Screenshot 2024-08-07 at 11 59 53 AM" src="https://github.com/user-attachments/assets/6acc80e7-9225-46b1-9180-4b67cddae1e2">

The trigrid budget is the best I could do. Initially I was using merge 2. for the trigrid and had the land running on a 0.9x1.25 FV grid. The annual heat budget was on the order of .26. It seems that the finer the land resolution - the better you do. So I switched to a 0.47x0.63 FV grid and moved to merge 1. That improved things but we are still an almost an order of magnitude greater than the bigrid heat budget.  Note that the heat budget for the bigrid is almost an order of magnitude smaller than the heat budget for the trigrid.

I have spent a lot of time using CESM2.1 and MCT to duplicate what E3SM did - but could not duplicate their current budget (which seems to be okay). They run the land on the rof grid at 1/2 degree. The main issue is that we don't really have the trigrid working for any land resolution and the budget even for 1/2 degree land is not acceptable at this point.

@adagj - can add more details regarding the analysis she did.

**Comment by adagj** [comment]
I analyzed the heat fluxes over land sent from CAM to the mediator and the heat fluxes received by CLM:

As an example, if we calculate the heat budgets for the longwave downwelling radiation in CAM using land (i.e., lfrac_a), ocean, and ice area fractions from CAM, they budgets sum up to the global budget for longwave downwelling radiation, as expected. However, when using lfin_a instead of lfrac, and comparing the values, the difference is nearly identical to the imbalance in the heat budget for downwelling longwave radiation calculated in the mediator, indicating that the issue lies with the difference fractions of land used by CAM and CLM (as Mariana already explained).

I also compared lfrin_a and lfrac_a and found that the edges where we have grid cells with fractional land and ocean are causing the imbalance in the energy budget in the mediator. The figure below shows the total area fraction using lfrin_a instead of lfrac_a.

![image](https://github.com/user-attachments/assets/dfd61b78-3797-4091-acdb-4e3c55be0d45)
